aten/src/ATen/core/ivalue_inl.h

```diff
@@ -522,6 +528,25 @@ c10::optional<T> generic_to(
   return std::move(ivalue).to<T>();
 }
 
+template <typename Tuple, std::size_t... I>
+Tuple generic_to_tuple_impl(
+    const std::vector<IValue>& t,
+    c10::guts::index_sequence<I...>) {
+  return std::make_tuple(
+      t[I].to<typename std::tuple_element<I, Tuple>::type>()...);
+}
+
+template <
+    typename... Args,
+    typename Indices = c10::guts::make_index_sequence<sizeof...(Args)>,
+    c10::guts::enable_if_t<
+        !c10::guts::disjunction<std::is_lvalue_reference<Args>...>::value,
+        std::nullptr_t> = nullptr>
+std::tuple<Args...> generic_to(IValue ivalue, _fake_type<std::tuple<Args...>>) {
+  auto vals = ivalue.toTuple()->elements();
```

Comment:
There should be a runtime test that the number of elements in the tuple matches the number of arguments in the parameter pack. I don't see that test in this function, or the function above.